src/Symfony/Component/HttpFoundation/UriSigner.php

```diff
@@ -104,6 +132,24 @@ private function buildUrl(array $url, array $params = []): string
 
         return $scheme.$user.$pass.$host.$port.$path.$query.$fragment;
     }
+
+    private function getExpirationTime(\DateTimeInterface|\DateInterval|int $expiration): string
+    {
+        if ($expiration instanceof \DateTimeInterface) {
+            return $expiration->format('U');
+        }
+
+        if ($expiration instanceof \DateInterval) {
+            return (new \DateTimeImmutable())->add($expiration)->format('U');
+        }
+
+        return (string) $expiration;
+    }
+
+    private function isExpired(int $expiration): bool
+    {
+        return time() >= $expiration;
```

Comment:
Maybe using Clock component optionally would be better with fallback, to allow using mock clock in tests for devs?

Reply:
It will create a dependency between HttpFoundation and the Clock component. How would you do it ?

Reply:
Require-dev? :)

Reply:
Mmmh I see, and in the `check` method I should get the current time like that ? 

```php
if (class_exists(\Symfony\Component\Clock\Clock::class)) {
            $time =  \Symfony\Component\Clock\Clock::get()->now()->format('U');
        } else {
            $time = time();
        }
```

Reply:
Probably. Or via DI, not sure what maintainers would prefer

Reply:
I wouldn't use the Clock component in this PR. Let's use time() for now and discuss for Clock applied to HttpFoundation